src/state/haproxy_route.py

```diff
@@ -0,0 +1,294 @@
+            tls_information: The charm's TLS information state component.
+            peers: List of IP address of haproxy peer units.
+
+        Raises:
+            HaproxyRouteIntegrationDataValidationError: When data validation failed.
+
+        Returns:
+            HaproxyRouteRequirersInformation: Information about requirers
+                for the haproxy-route interface.
+        """
+        try:
+            # This is used to check that requirers don't ask for the same backend name.
+            backend_names: set[str] = set()
+            # Control stick tables for rate_limiting and
+            # eventually any shared values between haproxy units.
+            stick_table_entries: list[str] = []
+            requirers = haproxy_route.get_data(haproxy_route.relations)
+            backends: list[HAProxyRouteBackend] = []
+            for requirer in requirers.requirers_data:
+                if requirer.application_data.service in backend_names:
+                    logger.error("Requirers requested duplicate backend names.")
+                    raise HaproxyRouteIntegrationDataValidationError(
+                        "Requirers requested duplicate backend names."
+                    )
+                backend_names.add(requirer.application_data.service)
+
+                if requirer.application_data.rate_limit:
+                    stick_table_entries.append(f"{requirer.application_data.service}_rate_limit")
+
+                backend = HAProxyRouteBackend(
+                    relation_id=requirer.relation_id,
+                    application_data=requirer.application_data,
+                    servers=get_servers_definition_from_requirer_data(requirer),
+                    external_hostname=tls_information.external_hostname,
+                )
+                backends.append(backend)
+
+            return HaproxyRouteRequirersInformation(
+                backends=backends,
+                stick_table_entries=stick_table_entries,
+                peers=list(map(lambda x: cast(IPvAnyAddress, x), peers)),
+            )
+        except DataValidationError as exc:
+            # This exception is only raised if the provider has "raise_on_validation_error" set
+            logger.error("Error validating requirer data: %s", str(exc))
+            raise HaproxyRouteIntegrationDataValidationError from exc
+
+    @model_validator(mode="after")
+    def check_backend_paths(self) -> Self:
+        """Output a warning if requirers declared conflicting paths/subdomains.
+
+        Returns:
+            Self: The validated model.
+        """
+        requirers_paths: list[str] = []
+        requirers_hostnames: list[str] = []
+
+        for backend in self.backends:
+            requirers_paths.extend(backend.application_data.paths)
+            requirers_hostnames.extend(backend.hostname_acls)
+
+        if len(requirers_paths) != len(set(requirers_paths)):
+            logger.error(
```

Comment:
If you don't plan to block under this scenario, this should probably be a warning

Reply:
Thanks, I was thinking to use `error` here to highlight that this is quite important and the routing will probably get messed up. changed it to warning.